apps/web/src/lib/inngest/process-bookmark.job.ts

```diff
@@ -75,23 +88,33 @@ export const processBookmarkJob = inngest.createFunction(
       });
     });
 
-    await step.run("update-bookmark-status", async () => {
-      await prisma.bookmark.update({
-        where: { id: bookmarkId },
-        data: { inngestRunId: runId, status: "PROCESSING" },
-      });
-    });
-
     if (!bookmark) {
       throw new Error("Bookmark not found");
     }
 
+    await step.run("update-bookmark-status", async () => {
+      await Promise.all([
+        prisma.bookmark.update({
+          where: { id: bookmarkId },
+          data: { inngestRunId: runId, status: "PROCESSING" },
+        }),
+        prisma.bookmarkProcessingRun.create({
+          data: {
+            inngestRunId: runId,
+            bookmarkId: bookmarkId,
+            userId: bookmark!.userId,
+            status: "STARTED",
+          },
+        }),
+      ]);
+    });
+
     // Validate bookmark limits before processing
     await step.run("validate-bookmark-limits", async () => {
       try {
         await validateBookmarkLimits({
-          userId: bookmark.userId,
-          url: bookmark.url,
+          userId: bookmark!.userId,
+          url: bookmark!.url,
```

Comment:
find a way to not validate like that, you should have bookmark required at this stp